apps/src/lib/kits/maker/WebSerialPortWrapper.js

```diff
@@ -0,0 +1,62 @@
+import {EventEmitter} from 'events';
+import {SERIAL_BAUD} from '@cdo/apps/lib/kits/maker/util/boardUtils';
+
+export default class WebSerialPortWrapper extends EventEmitter {
+  constructor(port) {
+    super();
+    this.port = port;
+    this.portOpen = false;
+
+    if (port) {
+      const portInfo = port.getInfo();
+      this.vendorId = portInfo.usbVendorId;
+      this.productId = portInfo.usbProductId;
+    }
+
+    this.writer = null;
+    this.reader = null;
+  }
+
+  // Return a list of available ports
+  list() {
+    // TODO - not sure if this is used in Maker Toolkit yet
+  }
+
+  // Opens the serial port and starts reading from port.
+  async open() {
+    if (this.portOpen) {
+      throw new Error(`Requested port is already open.`);
+    }
+
+    Promise.resolve()
+      .then(() => {
+        return this.port.open({baudRate: SERIAL_BAUD});
+      })
```

Comment:
Hm I'm a little confused by the use of promises here. Since we're using `async` I thought we might want to follow an `async / await` pattern, but then I realized here we do a `Promise.resolve()` to start a chain of `.then()`'s . I probably just don't have the context for why we need it to be this way. Can you explain a bit?

Reply:
Thanks for pointing this out! I removed the promised and added the "thens" to the end of `this.port.open` and it cleaned this up nicely!